clang-tools-extra/clangd/ConfigCompile.cpp

```diff
@@ -572,32 +572,43 @@ struct FragmentCompiler {
 #else
     static llvm::Regex::RegexFlags Flags = llvm::Regex::NoFlags;
 #endif
-    auto Filters = std::make_shared<std::vector<llvm::Regex>>();
-    for (auto &HeaderPattern : F.IgnoreHeader) {
-      // Anchor on the right.
-      std::string AnchoredPattern = "(" + *HeaderPattern + ")$";
-      llvm::Regex CompiledRegex(AnchoredPattern, Flags);
-      std::string RegexError;
-      if (!CompiledRegex.isValid(RegexError)) {
-        diag(Warning,
-             llvm::formatv("Invalid regular expression '{0}': {1}",
-                           *HeaderPattern, RegexError)
-                 .str(),
-             HeaderPattern.Range);
-        continue;
+    std::shared_ptr<std::vector<llvm::Regex>> Filters;
+    if (!F.IgnoreHeader.empty()) {
```

Comment:
nit: you can keep this part the same, and just replace `if (Filters->emtpy()) return;` with `Filters.reset();`

Reply:
But then if `F.IgnoreHeader` is empty, we'll be allocating and deallocating an unneeded vector.

It's also possible to use `if (Filters->empty() && !AnalyzeSystemHeaders.has_value()) return;` and check `Filters->empty()`again  in line 601, but I wanted to remove the unneeded allocation.

Reply:
config compilation isn't really a hot code path, hence i'd lean towards simplicity. up to you though.